cli/src/main/java/org/infinispan/cli/Context.java

```diff
@@ -72,12 +78,20 @@
     */
    void disconnect();
 
-   CommandResult execute(Shell shell, List<CommandInputLine> commands);
+   CommandResult execute(Shell shell, Function<RestClient, CompletionStage<RestResponse>> op, Connection.ResponseMode responseMode);
+
+   CommandResult execute(Shell shell, BiFunction<RestClient, Resource, CompletionStage<RestResponse>> op, Connection.ResponseMode responseMode);
```

Comment:
Just 2 notes before starting to review this:

* Wouldn't be better/simpler just to add an abstract method to `CliCommand`? like `abstract CompletionStage<RestResponse> invokeInServer(RestClient client, Resource resource);`

* Also, I would remove the `execute` overload and use `Resource` as parameter since there is no difference between `execute` implementations.

Reply:
I've done that. I've moved the commands around a bit too